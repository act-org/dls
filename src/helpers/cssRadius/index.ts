/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

import { StandardCSSProperties } from '@mui/system';

import px from '~/helpers/px';

type BorderRadius = Pick<
  StandardCSSProperties,
  | 'borderBottomLeftRadius'
  | 'borderBottomRightRadius'
  | 'borderRadius'
  | 'borderTopLeftRadius'
  | 'borderTopRightRadius'
>;

export const cssRadius = (value: number): BorderRadius => ({
  borderBottomLeftRadius: px(value),
  borderBottomRightRadius: px(value),
  borderRadius: px(value),
  borderTopLeftRadius: px(value),
  borderTopRightRadius: px(value),
});

export default cssRadius;
