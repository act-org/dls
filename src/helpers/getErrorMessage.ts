/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { get } from 'lodash';

export const getErrorMessage = (err: unknown): string => {
  let errorMessage = '';
  const graphQlErrorMessage = get(err, 'networkError.result.errors[0].message');
  const standardErrorMessage = get(err, 'message');

  if (graphQlErrorMessage) {
    errorMessage = graphQlErrorMessage;
  } else if (standardErrorMessage) {
    errorMessage = standardErrorMessage;
  }

  return errorMessage.trim();
};

export default getErrorMessage;
