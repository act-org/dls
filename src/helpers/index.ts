/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

export * from './cssRadius';
export * from './getErrorMessage';
export * from './makeShadow';
export * from './mergeClasses';
export * from './oxfordCommaJoin';
export * from './px';
export * from './pxToNumber';
export * from './search';
export * from './sort';
export * from './styled';
export * from './types';
