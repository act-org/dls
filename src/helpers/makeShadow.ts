/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

import px from '@actinc/dls/helpers/px';

export const makeShadow = (
  xOffset: number,
  yOffset: number,
  blurRadius: number | undefined,
  spreadRadius: number | undefined,
  color: string,
  inset = false,
): string => {
  return `${inset ? 'inset ' : ''}${px(xOffset)} ${px(yOffset)} ${`${px(
    blurRadius,
  )} `}${`${px(spreadRadius)} `}${color}`;
};

export default makeShadow;
