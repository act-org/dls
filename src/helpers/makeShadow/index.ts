/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import isNumber from 'lodash/isNumber';

import px from '~/helpers/px';

export const makeShadow = (
  xOffset: number,
  yOffset: number,
  blurRadius: number | undefined,
  spreadRadius: number | undefined,
  color: string,
  inset = false,
): string => {
  let str = '';

  if (inset) {
    str = `${str} inset`;
  }

  if (isNumber(xOffset)) {
    str = `${str} ${px(xOffset)}`;
  }

  if (isNumber(yOffset)) {
    str = `${str} ${px(yOffset)}`;
  }

  if (isNumber(blurRadius)) {
    str = `${str} ${px(blurRadius)}`;
  }

  if (isNumber(spreadRadius)) {
    str = `${str} ${px(spreadRadius)}`;
  }

  if (color) {
    str = `${str} ${color}`;
  }

  return str.trim();
};

export default makeShadow;
