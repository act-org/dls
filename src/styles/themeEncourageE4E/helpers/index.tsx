/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

export type ColorIndex =
  | 50
  | 100
  | 200
  | 300
  | 400
  | 500
  | 600
  | 700
  | 800
  | 900
  | string;

interface IBrightnessMap {
  brightness: number;
  colorIndex: ColorIndex;
}

/*
 * Do not modify these values unless UX give us new values,
 * they are used to generate our Figma palette
 */
const brightnessCoefficient: IBrightnessMap[] = [
  {
    brightness: 0.9,
    colorIndex: 50,
  },
  {
    brightness: 0.69,
    colorIndex: 100,
  },
  {
    brightness: 0.54,
    colorIndex: 200,
  },
  {
    brightness: 0.33,
    colorIndex: 300,
  },
  {
    brightness: 0.2,
    colorIndex: 400,
  },
  {
    brightness: 0,
    colorIndex: 500,
  },
  {
    brightness: -0.09,
    colorIndex: 600,
  },
  {
    brightness: -0.29,
    colorIndex: 700,
  },
  {
    brightness: -0.45,
    colorIndex: 800,
  },
  {
    brightness: -0.58,
    colorIndex: 900,
  },
];

export const hexToRGB = (color: string): number[] | undefined => {
  let r = '';
  let g = '';
  let b = '';

  if ((color.length !== 4 && color.length !== 7) || color[0] !== '#')
    return undefined;

  if (color.length === 4) {
    r = `0x${color[1]}${color[1]}`;
    g = `0x${color[2]}${color[2]}`;
    b = `0x${color[3]}${color[3]}`;
  } else if (color.length === 7) {
    r = `0x${color[1]}${color[2]}`;
    g = `0x${color[3]}${color[4]}`;
    b = `0x${color[5]}${color[6]}`;
  }

  return [Number(r), Number(g), Number(b)];
};

export const RGBToHex = (r: number, g: number, b: number): string => {
  let R = r.toString(16);
  let G = g.toString(16);
  let B = b.toString(16);

  if (R.length === 1) R = `0${R}`;
  if (G.length === 1) G = `0${G}`;
  if (B.length === 1) B = `0${B}`;

  return `#${R.slice(0, 2)}${G.slice(0, 2)}${B.slice(0, 2)}`;
};

export const lightenRGB = (
  color: string,
  percentage: number,
): string | undefined => {
  const RGBColor = hexToRGB(color);

  if (!RGBColor) return undefined;

  if (percentage === 0) return color;

  if (percentage > 0) {
    return RGBToHex(
      Math.round(RGBColor[0] + ((255 - RGBColor[0]) * percentage) / 100),
      Math.round(RGBColor[1] + ((255 - RGBColor[1]) * percentage) / 100),
      Math.round(RGBColor[2] + ((255 - RGBColor[2]) * percentage) / 100),
    );
  }

  return RGBToHex(
    Math.round(RGBColor[0] * (1 + percentage / 100)),
    Math.round(RGBColor[1] * (1 + percentage / 100)),
    Math.round(RGBColor[2] * (1 + percentage / 100)),
  );
};

export const figmaColorGenerator = (
  color: string,
): Record<ColorIndex, string> => {
  const colorPalette = {} as unknown as Record<ColorIndex, string>;
  brightnessCoefficient.forEach(({ brightness, colorIndex }) => {
    const generatedColor = lightenRGB(color, brightness * 100);
    if (generatedColor) colorPalette[colorIndex] = generatedColor;
  });
  return colorPalette;
};

export const generatePaletteColors = <
  T extends { colorName: U; colorHEX: string },
  U extends string,
>(
  colors: T[],
): Record<U, Record<ColorIndex, string>> => {
  const colorPalette = {} as unknown as Record<U, Record<ColorIndex, string>>;
  colors.forEach(color => {
    colorPalette[color.colorName] = figmaColorGenerator(color.colorHEX);
  });
  return colorPalette;
};
