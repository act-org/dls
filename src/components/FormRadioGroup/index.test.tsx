/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

import * as React from 'react';

import { standard } from '~/helpers/test';

import { FormRadioGroup } from '.';

jest.mock('@mui/material/utils/useId', () => {
  let initialId = 12345;
  return jest.fn(() => {
    // eslint-disable-next-line no-plusplus
    return `mui-${initialId++}`;
  });
});

describe('FormRadioGroup', () => {
  const Component = (
    <FormRadioGroup
      id="primaryradiogroup"
      label="The label"
      name="radiooptions"
      options={[
        { id: 'one', label: 'one', value: 'one' },
        { id: 'two', label: 'two', value: 'two' },
      ]}
    />
  );
  standard(Component);
});
