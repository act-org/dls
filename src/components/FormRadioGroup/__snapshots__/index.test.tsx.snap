// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormRadioGroup ACT theme matches the snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1o36m94-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
    >
      <div
        class="label-split"
      >
        <span
          class="label-help"
        >
          The label
        </span>
      </div>
    </label>
    <div
      aria-label="radiooptions"
      class="MuiFormGroup-root MuiRadioGroup-root css-19fxcl9-MuiFormGroup-root"
      id="primaryradiogroup"
      role="radiogroup"
    >
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-mkv5eu-MuiFormControlLabel-root"
        id="one"
      >
        <span
          class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-1ylxud-MuiButtonBase-root-MuiRadio-root"
          color="primary"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            name=":r0:"
            type="radio"
            value="one"
          />
          <span
            class="css-hyxlzm"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lgwimo-MuiSvgIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-nrr40h-MuiSvgIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-risc0p-MuiTypography-root"
        >
          one
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-mkv5eu-MuiFormControlLabel-root"
        id="two"
      >
        <span
          class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-1ylxud-MuiButtonBase-root-MuiRadio-root"
          color="primary"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            name=":r0:"
            type="radio"
            value="two"
          />
          <span
            class="css-hyxlzm"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lgwimo-MuiSvgIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-nrr40h-MuiSvgIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-risc0p-MuiTypography-root"
        >
          two
        </span>
      </label>
    </div>
  </div>
</div>
`;

exports[`FormRadioGroup ACT_ET theme matches the snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-osbo88-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
    >
      <div
        class="label-split"
      >
        <span
          class="label-help"
        >
          The label
        </span>
      </div>
    </label>
    <div
      aria-label="radiooptions"
      class="MuiFormGroup-root MuiRadioGroup-root css-1mac7pt-MuiFormGroup-root"
      id="primaryradiogroup"
      role="radiogroup"
    >
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1w7u07-MuiFormControlLabel-root"
        id="one"
      >
        <span
          class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-60abqe-MuiButtonBase-root-MuiRadio-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            name=":r1:"
            type="radio"
            value="one"
          />
          <span
            class="css-hyxlzm"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-c8798c-MuiSvgIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-g7uato-MuiSvgIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-1b9k1hy-MuiTypography-root"
        >
          one
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1w7u07-MuiFormControlLabel-root"
        id="two"
      >
        <span
          class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-60abqe-MuiButtonBase-root-MuiRadio-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            name=":r1:"
            type="radio"
            value="two"
          />
          <span
            class="css-hyxlzm"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-c8798c-MuiSvgIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-g7uato-MuiSvgIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-1b9k1hy-MuiTypography-root"
        >
          two
        </span>
      </label>
    </div>
  </div>
</div>
`;

exports[`FormRadioGroup ENCOURA theme matches the snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-pta9t1-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
    >
      <div
        class="label-split"
      >
        <span
          class="label-help"
        >
          The label
        </span>
      </div>
    </label>
    <div
      aria-label="radiooptions"
      class="MuiFormGroup-root MuiRadioGroup-root css-1mac7pt-MuiFormGroup-root"
      id="primaryradiogroup"
      role="radiogroup"
    >
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-mkv5eu-MuiFormControlLabel-root"
        id="one"
      >
        <span
          class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-1wptl0z-MuiButtonBase-root-MuiRadio-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            name=":r2:"
            type="radio"
            value="one"
          />
          <span
            class="css-hyxlzm"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-4ei384-MuiSvgIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-tl6vf7-MuiSvgIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-kiofek-MuiTypography-root"
        >
          one
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-mkv5eu-MuiFormControlLabel-root"
        id="two"
      >
        <span
          class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-1wptl0z-MuiButtonBase-root-MuiRadio-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            name=":r2:"
            type="radio"
            value="two"
          />
          <span
            class="css-hyxlzm"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-4ei384-MuiSvgIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-tl6vf7-MuiSvgIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-kiofek-MuiTypography-root"
        >
          two
        </span>
      </label>
    </div>
  </div>
</div>
`;

exports[`FormRadioGroup ENCOURA_CLASSIC theme matches the snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1291igl-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
    >
      <div
        class="label-split"
      >
        <span
          class="label-help"
        >
          The label
        </span>
      </div>
    </label>
    <div
      aria-label="radiooptions"
      class="MuiFormGroup-root MuiRadioGroup-root css-1mac7pt-MuiFormGroup-root"
      id="primaryradiogroup"
      role="radiogroup"
    >
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1w7u07-MuiFormControlLabel-root"
        id="one"
      >
        <span
          class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-19o6bw-MuiButtonBase-root-MuiRadio-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            name=":r3:"
            type="radio"
            value="one"
          />
          <span
            class="css-hyxlzm"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-4ei384-MuiSvgIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-tl6vf7-MuiSvgIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-kiofek-MuiTypography-root"
        >
          one
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1w7u07-MuiFormControlLabel-root"
        id="two"
      >
        <span
          class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-19o6bw-MuiButtonBase-root-MuiRadio-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            name=":r3:"
            type="radio"
            value="two"
          />
          <span
            class="css-hyxlzm"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-4ei384-MuiSvgIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-tl6vf7-MuiSvgIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-kiofek-MuiTypography-root"
        >
          two
        </span>
      </label>
    </div>
  </div>
</div>
`;

exports[`FormRadioGroup ENCOURAGE theme matches the snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root css-jayakb-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-bczahb-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
    >
      <div
        class="label-split"
      >
        <span
          class="label-help"
        >
          The label
        </span>
      </div>
    </label>
    <div
      aria-label="radiooptions"
      class="MuiFormGroup-root MuiRadioGroup-root css-1mac7pt-MuiFormGroup-root"
      id="primaryradiogroup"
      role="radiogroup"
    >
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1s9jovc-MuiFormControlLabel-root"
        id="one"
      >
        <span
          class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-nlfcay-MuiButtonBase-root-MuiRadio-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            name=":r4:"
            type="radio"
            value="one"
          />
          <span
            class="css-hyxlzm"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-c8798c-MuiSvgIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-g7uato-MuiSvgIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-evbvr7-MuiTypography-root"
        >
          one
        </span>
      </label>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1s9jovc-MuiFormControlLabel-root"
        id="two"
      >
        <span
          class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-nlfcay-MuiButtonBase-root-MuiRadio-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            name=":r4:"
            type="radio"
            value="two"
          />
          <span
            class="css-hyxlzm"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-c8798c-MuiSvgIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-g7uato-MuiSvgIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-evbvr7-MuiTypography-root"
        >
          two
        </span>
      </label>
    </div>
  </div>
</div>
`;
